merjdv-z170411/task06.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+"""Task06.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1nyR-tTtqy8h8yv5vyeslI7oljBQElfoi
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""Read the RDF file as shown in class"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example5.rdf", format="xml")
+
+"""Create a new class named Researcher"""
+
+ns = Namespace("http://somewhere#")
+g.add((ns.Researcher, RDF.type, RDFS.Class))
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.1: Create a new class named "University"**
+
+"""
+
+# TO DO
+# Visualize the results
+g.add((ns.University, RDF.type, RDFS.class))
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.2: Add "Researcher" as a subclass of "Person"**"""
+
+# TO DO
+# Visualize the results
+g.add((ns.Researcher, RDFS.subClassOf, ns.Person))
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.3: Create a new individual of Researcher named "Jane Smith"**"""
+
+# TO DO
+# Visualize the results
+jSmith = ns.JaneSmith
+g.add((jSmith, RDF.type, ns.Researcher))
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.4: Add to the individual JaneSmith the email address, fullName, given and family names**"""
+
+# TO DO
+# Visualize the results
+g.add((jSmith, ns.email, Literal("janesm@example.com")))
+g.add((jSmith, ns.fullName, Literal("Jane Smith")))
+g.add((jSmith, ns.givenName, Literal("Jane")))
+g.add((jSmith, ns.familyName, Literal("Smith")))
+
+for s, p, o in g:
+  print(s,p,o)
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# TO DO
+# Visualize the results
+joSmith = ns.JohnSmith
+upm = ns.UPM
+
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((joSmith, ns.worksAt, upm))
+
+for s, p, o in g.triples((joSmith,ns.worksAt,None)):
+  print(s,p,o)
+
+"""**Task 6.6: Add that Jown knows Jane using the FOAF vocabulary**"""
+
+# TO DO
+# Visualize the results
+
+FOAF = Namespace("http://xmlns.com/foaf/0.1/")
+
+john = Namespace("http://somewhere#John")
+jane = Namespace("http://somewhere#Jane")
```

Comment:
No need to use different namespaces. Use 1 and then do john = ns.John as shown in class